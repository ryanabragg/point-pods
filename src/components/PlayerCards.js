import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';

import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';

const styles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignItems: 'flex-start',
  },
  card: {
    minWidth: 275,
    margin: theme.spacing.unit,
  },
  right: {
    marginLeft: 'auto',
  },
});

class PlayerCards extends Component {
  render() {
    const { classes, search, players } = this.props;
    let cards = !search
      ? players
      : players.filter(t => t.name.toLowerCase().includes(search));
    return (
      <div className={classes.root}>
        {cards.map(({id, name, points}) => (
          <Card key={id} id={id} className={classes.card}>
            <CardHeader title={name} />
            <CardContent>
              <Typography>
                {`${points} Cumulative Points`}
              </Typography>
            </CardContent>
            <CardActions>
              <Button size='small'
                className={classes.right}
                href={`/player/${id}`}
              >
                Details
              </Button>
            </CardActions>
          </Card>
        ))}
        {!cards.length && (
          <Card id='not-found' className={classes.card}>
            <CardHeader
              title='No Players Found'
            />
          </Card>
        )}
      </div>
    );
  }
}

PlayerCards.defaultProps = {
  players: [],
  selected: [],
  onSelect: (tournaments) => null,
  search: '',
};

PlayerCards.propTypes = {
  classes: PropTypes.object.isRequired, // added by withStyles
  players: PropTypes.array,
  selected: PropTypes.array,
  onSelect: PropTypes.func,
  search: PropTypes.string,
};

export default withStyles(styles, { withTheme: true })(PlayerCards);
