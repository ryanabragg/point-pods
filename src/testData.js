import api from './api';

const testSettings = {
  pairingMethod: api.pairingMethods.HISTORIC,
  pairingMethodInitial: api.pairingMethods.HISTORIC,
  podSizeMinimum: 4,
  podSizeMaximum: 7,
};

const testPlayers = [{
  id: 'mock0',
  name: 'test0',
  points: 0,
}, {
  id: 'mock1',
  name: 'test1',
  points: 1,
}, {
  id: 'mock2',
  name: 'test2',
  points: 2,
}, {
  id: 'mock3',
  name: 'test3',
  points: 3,
}, {
  id: 'mock4',
  name: 'test4',
  points: 4,
}, {
  id: 'mock5',
  name: 'test5',
  points: 5,
}, {
  id: 'mock6',
  name: 'test6',
  points: 6,
}, {
  id: 'mock7',
  name: 'test7',
  points: 7,
}, {
  id: 'mock8',
  name: 'test8',
  points: 8,
}, {
  id: 'mock9',
  name: 'test9',
  points: 9,
}];

const testTournaments = [{
  id: 'mock0',
  name: 'test0',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Etiam urna tortor, laoreet vitae commodo porttitor, mattis id purus.',
  date: new Date('2018-01-01'),
  done: false,
  staging: true,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 3,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 3 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 0,
    participated: false,
    dropped: false,
  }],
}, {
  id: 'mock1',
  name: 'test1',
  category: 'cat1',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed molestie suscipit suscipit. Nullam sollicitudin tempus pretium.',
  date: new Date('2018-02-01'),
  done: true,
  staging: false,
  pairingMethod: api.pairingMethods.POINTS,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 5,
  rounds: 2,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 0 },
    2: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: true,
    1: { pod: 1, points: 1 },
    2: { pod: 2, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
    2: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 3 },
    2: { pod: 2, points: 3 },
  }, {
    id: 'mock4',
    name: 'test4',
    points: 4,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 4 },
    2: { pod: 1, points: 4 },
  }, {
    id: 'mock5',
    name: 'test5',
    points: 5,
    participated: true,
    dropped: false,
    1: { pod: 2, points: 5 },
    2: { pod: 2, points: 5 },
  }, {
    id: 'mock6',
    name: 'test6',
    points: 6,
    participated: true,
    dropped: false,
    2: { pod: 2, points: 6 },
  }],
}, {
  id: 'mock2',
  name: 'test2',
  category: 'cat0',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae tellus nunc. Sed ullamcorper metus vitae nisi rhoncus placerat.',
  date: new Date('2018-01-01'),
  done: false,
  staging: false,
  pairingMethod: api.pairingMethods.RANDOM,
  pairingMethodInitial: api.pairingMethods.RANDOM,
  podSizeMinimum: 3,
  podSizeMaximum: 4,
  rounds: 1,
  players: [{
    id: 'mock0',
    name: 'test0',
    points: 0,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 0 },
  }, {
    id: 'mock1',
    name: 'test1',
    points: 1,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 1 },
  }, {
    id: 'mock2',
    name: 'test2',
    points: 2,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 2 },
  }, {
    id: 'mock3',
    name: 'test3',
    points: 3,
    participated: true,
    dropped: false,
    1: { pod: 1, points: 3 },
  }],
}];

const mockRoute = {
  history: {
    push: jest.fn(to => to),
    replace: jest.fn(to => to),
  },
  match: {
    params: {},
  },
  matchWithInvalid: {
    params: {
      id: 'invalid',
    },
  },
  matchWithID0: {
    params: {
      id: 'mock0',
    },
  },
  matchWithID1: {
    params: {
      id: 'mock1',
    },
  },
};

let mockStore = {};
const mockStorage = {
  setItem: jest.fn((key, val) => Promise.resolve((mockStore[key] = val))),
  getItem: jest.fn(key => Promise.resolve(mockStore[key])),
  removeItem: jest.fn(key => { delete mockStore[key]; Promise.resolve(); }),
  clear: jest.fn(() => Promise.resolve((mockStore = {}))),
  keys: jest.fn(() => Promise.resolve(Object.keys(mockStore))),
  length: jest.fn(() => Promise.resolve(Object.keys(mockStore).length)),
};

let defaultStore = null;
function mockAPI(api, storage) {
  defaultStore = api.store;
  api.store = storage;
  mockStore = {
    settings: testSettings,
    players: testPlayers,
    tournaments: testTournaments,
  };
}
function demockAPI(api) {
  api.store = defaultStore;
  defaultStore = null;
}

export {
  testSettings,
  testPlayers,
  testTournaments,
  mockRoute,
  mockStore,
  mockStorage,
  mockAPI,
  demockAPI,
};
