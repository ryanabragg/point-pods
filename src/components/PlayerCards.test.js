import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Card from '@material-ui/core/Card';

import PlayerCards from './PlayerCards';

import { testPlayers } from '../testData';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PlayerCards />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('matches the prior snapshot', () => {
  const component = renderer.create(<PlayerCards />);
  expect(component.toJSON()).toMatchSnapshot();
  component.update(<PlayerCards players={testPlayers} />);
  expect(component.toJSON()).toMatchSnapshot();
});

test('selection');

describe('filter props', () => {
  test('search', () => {
    const search = [{
      term: 'test2',
      expect: [ 'mock2' ],
    }, {
      term: 'tes',
      expect: [ 'mock0', 'mock1', 'mock2', 'mock3', 'mock4', 'mock5', 'mock6', 'mock7', 'mock8', 'mock9' ],
    }, {
      term: '5',
      expect: [ 'mock5' ],
    }];
    const wrapper = shallow(<PlayerCards players={testPlayers} />);
    const component = wrapper.dive();
    expect(component.find(Card).length).toBe(testPlayers.length);
    search.forEach((find, index) => {
      component.setProps({
        search: find.term,
      });
      expect(component.find(Card).length).toBe(find.expect.length);
      component.find(Card).forEach((node, i) => {
        expect(node.prop('id')).toBe(find.expect[i]);
      });
    });
  });

  test('no results', () => {
    const wrapper = shallow(<PlayerCards players={testPlayers} />);
    const component = wrapper.dive();
    expect(component.find(Card).length).toBe(testPlayers.length);
    component.setProps({
      search: 'not-found'
    });
    expect(component.find(Card).length).toBe(1);
    expect(component.find(Card).prop('id')).toBe('not-found');
  });
});
