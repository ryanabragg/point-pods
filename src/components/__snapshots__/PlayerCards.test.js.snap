// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the prior snapshot 1`] = `
<div
  className="PlayerCards-root-63"
>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="not-found"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          No Players Found
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`matches the prior snapshot 2`] = `
<div
  className="PlayerCards-root-63"
>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock0"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test0
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        0 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock0"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock1"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test1
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        1 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock1"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock2"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test2
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        2 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock2"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock3"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test3
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        3 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock3"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock4"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test4
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        4 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock4"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock5"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test5
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        5 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock5"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock6"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test6
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        6 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock6"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock7"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test7
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        7 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock7"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock8"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test8
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        8 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock8"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
  <div
    className="MuiPaper-root-67 MuiPaper-elevation1-70 MuiPaper-rounded-68 MuiCard-root-66 PlayerCards-card-64"
    id="mock9"
  >
    <div
      className="MuiCardHeader-root-94"
    >
      <div
        className="MuiCardHeader-content-97"
      >
        <span
          className="MuiTypography-root-100 MuiTypography-headline-105 MuiCardHeader-title-98"
        >
          test9
        </span>
      </div>
    </div>
    <div
      className="MuiCardContent-root-125"
    >
      <p
        className="MuiTypography-root-100 MuiTypography-body1-109"
      >
        9 Cumulative Points
      </p>
    </div>
    <div
      className="MuiCardActions-root-126"
    >
      <a
        className="MuiButtonBase-root-152 MuiButton-root-128 MuiButton-text-130 MuiButton-flat-133 MuiButton-sizeSmall-149 PlayerCards-right-65 MuiCardActions-action-127"
        href="/player/mock9"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <span
          className="MuiButton-label-129"
        >
          Details
        </span>
        <span
          className="MuiTouchRipple-root-155"
        />
      </a>
    </div>
  </div>
</div>
`;
